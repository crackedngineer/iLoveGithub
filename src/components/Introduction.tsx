type BadgeType = {
  href: string;
  src: string;
  alt: string;
};

export const Introduction = () => {
  const badges: BadgeType[] = [
    {
      href: "https://www.producthunt.com/posts/ilovegithub?embed=true&utm_source=badge-featured&utm_medium=badge&utm_source=badge-ilovegithub",
      src: "https://api.producthunt.com/widgets/embed-image/v1/featured.svg?post_id=967157&theme=light&t=1747590463334",
      alt: "iLoveGithub Product Hunt Badge",
    },
    {
      href: "https://peerlist.io/crackedngineer/project/ilovegithub",
      src: "https://peerlist.io/api/v1/projects/embed/PRJH8OEJ9OJ8E88PJ3R66977NBDAOK?showUpvote=true&theme=light",
      alt: "iLoveGithub Peerlist Badge",
    },
    {
      href: "https://scoutforge.net/reviews/ilovegithub/",
      src: "https://scoutforge.net/wp-content/themes/wp-theme/assets/img/badges/badge-light.webp",
      alt: "Trusted and Reviewed by Scout Forge Badge",
    },
  ];

  return (
    <div className="text-center mb-10 px-2 sm:px-4">
      {/* Badge Container without Background */}
      <div className="flex flex-wrap justify-center gap-4 p-2">
        {badges.map((badge: BadgeType, index) => (
          <a
            key={index}
            href={badge.href}
            target="_blank"
            rel="noopener noreferrer"
            className="transform hover:scale-105 transition duration-300"
          >
            {
              // eslint-disable-next-line @next/next/no-img-element
              <img src={badge.src} alt={badge.alt} className="rounded-md h-14" width={250} />
            }
          </a>
        ))}
      </div>

      <h1 className="text-4xl md:text-5xl font-bold bg-gradient-to-r from-github-blue to-github-green bg-clip-text text-transparent mt-6 mb-4">
        Discover GitHub Tools
      </h1>
      <p className="text-lg md:text-xl text-gray-600 dark:text-gray-400 max-w-2xl mx-auto mb-4">
        Explore GitHub repositories and find the best tools to enhance your GitHub experience
      </p>
    </div>
  );
};
