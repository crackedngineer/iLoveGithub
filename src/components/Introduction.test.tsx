import {render, screen} from "@testing-library/react";
import {Introduction} from "./Introduction";

describe("Introduction Component", () => {
  it("renders the main heading", () => {
    render(<Introduction />);
    const heading = screen.getByRole("heading", {name: /discover github tools/i});
    expect(heading).toBeInTheDocument();
  });

  it("renders the description text", () => {
    render(<Introduction />);
    const description = screen.getByText(/explore github repositories and find the best tools/i);
    expect(description).toBeInTheDocument();
  });

  it("renders all three badge links", () => {
    render(<Introduction />);
    const links = screen.getAllByRole("link");
    expect(links).toHaveLength(3);
  });

  it("renders Product Hunt badge with correct attributes", () => {
    render(<Introduction />);
    const productHuntLink = screen.getByRole("link", {name: /ilovegithub product hunt badge/i});

    expect(productHuntLink).toHaveAttribute(
      "href",
      "https://www.producthunt.com/posts/ilovegithub?embed=true&utm_source=badge-featured&utm_medium=badge&utm_source=badge-ilovegithub",
    );
    expect(productHuntLink).toHaveAttribute("target", "_blank");
    expect(productHuntLink).toHaveAttribute("rel", "noopener noreferrer");
  });

  it("renders Peerlist badge with correct attributes", () => {
    render(<Introduction />);
    const peerlistLink = screen.getByRole("link", {name: /ilovegithub peerlist badge/i});

    expect(peerlistLink).toHaveAttribute(
      "href",
      "https://peerlist.io/crackedngineer/project/ilovegithub",
    );
    expect(peerlistLink).toHaveAttribute("target", "_blank");
    expect(peerlistLink).toHaveAttribute("rel", "noopener noreferrer");
  });

  it("renders Scout Forge badge with correct attributes", () => {
    render(<Introduction />);
    const scoutForgeLink = screen.getByRole("link", {
      name: /trusted and reviewed by scout forge badge/i,
    });

    expect(scoutForgeLink).toHaveAttribute("href", "https://scoutforge.net/reviews/ilovegithub/");
    expect(scoutForgeLink).toHaveAttribute("target", "_blank");
    expect(scoutForgeLink).toHaveAttribute("rel", "noopener noreferrer");
  });

  it("renders all badge images with correct alt text", () => {
    render(<Introduction />);

    const productHuntImg = screen.getByAltText(/ilovegithub product hunt badge/i);
    const peerlistImg = screen.getByAltText(/ilovegithub peerlist badge/i);
    const scoutForgeImg = screen.getByAltText(/trusted and reviewed by scout forge badge/i);

    expect(productHuntImg).toBeInTheDocument();
    expect(peerlistImg).toBeInTheDocument();
    expect(scoutForgeImg).toBeInTheDocument();
  });

  it("applies correct styling classes to badge images", () => {
    render(<Introduction />);
    const images = screen.getAllByRole("img");

    images.forEach((img) => {
      expect(img).toHaveAttribute("width", "250");
      expect(img).toHaveClass("rounded-md", "h-14");
    });
  });

  it("applies hover effect class to badge links", () => {
    render(<Introduction />);
    const links = screen.getAllByRole("link");

    links.forEach((link) => {
      expect(link).toHaveClass("transform", "hover:scale-105", "transition", "duration-300");
    });
  });

  it("applies gradient text styling to heading", () => {
    render(<Introduction />);
    const heading = screen.getByRole("heading", {name: /discover github tools/i});

    expect(heading).toHaveClass(
      "bg-gradient-to-r",
      "from-github-blue",
      "to-github-green",
      "bg-clip-text",
      "text-transparent",
    );
  });

  it("applies responsive text sizing to heading", () => {
    render(<Introduction />);
    const heading = screen.getByRole("heading", {name: /discover github tools/i});

    expect(heading).toHaveClass("text-4xl", "md:text-5xl");
  });

  it("applies responsive text sizing to description", () => {
    render(<Introduction />);
    const description = screen.getByText(/explore github repositories and find the best tools/i);

    expect(description).toHaveClass("text-lg", "md:text-xl");
  });

  it("renders badge container with flexbox layout", () => {
    const {container} = render(<Introduction />);
    const badgeContainer = container.querySelector(".flex.flex-wrap.justify-center");

    expect(badgeContainer).toBeInTheDocument();
    expect(badgeContainer).toHaveClass("gap-4", "p-2");
  });

  it("matches snapshot", () => {
    const {container} = render(<Introduction />);
    expect(container).toMatchSnapshot();
  });
});
