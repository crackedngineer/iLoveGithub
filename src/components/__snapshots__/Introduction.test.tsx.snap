// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Introduction Component matches snapshot 1`] = `
<div>
  <div
    class="text-center mb-10 px-2 sm:px-4"
  >
    <div
      class="flex flex-wrap justify-center gap-4 p-2"
    >
      <a
        class="transform hover:scale-105 transition duration-300"
        href="https://www.producthunt.com/posts/ilovegithub?embed=true&utm_source=badge-featured&utm_medium=badge&utm_source=badge-ilovegithub"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="iLoveGithub Product Hunt Badge"
          class="rounded-md h-14"
          src="https://api.producthunt.com/widgets/embed-image/v1/featured.svg?post_id=967157&theme=light&t=1747590463334"
          width="250"
        />
      </a>
      <a
        class="transform hover:scale-105 transition duration-300"
        href="https://peerlist.io/crackedngineer/project/ilovegithub"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="iLoveGithub Peerlist Badge"
          class="rounded-md h-14"
          src="https://peerlist.io/api/v1/projects/embed/PRJH8OEJ9OJ8E88PJ3R66977NBDAOK?showUpvote=true&theme=light"
          width="250"
        />
      </a>
      <a
        class="transform hover:scale-105 transition duration-300"
        href="https://scoutforge.net/reviews/ilovegithub/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Trusted and Reviewed by Scout Forge Badge"
          class="rounded-md h-14"
          src="https://scoutforge.net/wp-content/themes/wp-theme/assets/img/badges/badge-light.webp"
          width="250"
        />
      </a>
    </div>
    <h1
      class="text-4xl md:text-5xl font-bold bg-gradient-to-r from-github-blue to-github-green bg-clip-text text-transparent mt-6 mb-4"
    >
      Discover GitHub Tools
    </h1>
    <p
      class="text-lg md:text-xl text-gray-600 dark:text-gray-400 max-w-2xl mx-auto mb-4"
    >
      Explore GitHub repositories and find the best tools to enhance your GitHub experience
    </p>
  </div>
</div>
`;
